diff --git a/LLaVA/llava/model/multimodal_encoder/clip_encoder.py b/LLaVA/llava/model/multimodal_encoder/clip_encoder.py
index a54f326..2340381 100644
--- a/LLaVA/llava/model/multimodal_encoder/clip_encoder.py
+++ b/LLaVA/llava/model/multimodal_encoder/clip_encoder.py
@@ -117,7 +117,7 @@ class CLIPVisionTower(nn.Module):
 
         if if_adaptive:
             reduction_ratio = outlier_dectection(cls_attn)#*3.5
-        _, idx = torch.topk(cls_attn, int(N*reduction_ratio), dim=1, largest=True)  # [B, left_tokens] , sorted=True
+        _, idx = torch.topk(cls_attn, max(1, round(N * reduction_ratio)), dim=1, largest=True)  # [B, left_tokens] , sorted=True
         index = idx.unsqueeze(-1).expand(-1, -1, C)  # [B, left_tokens, C]
 
         Key_wo_cls = desired_layer_k[:, 1:]  # [B, N-1, C]
@@ -206,7 +206,7 @@ class CLIPVisionTower(nn.Module):
 
         if if_adaptive:
             reduction_ratio = outlier_dectection(cls_attn)#*3.5
-        _, idx = torch.topk(cls_attn, int(N*reduction_ratio), dim=1, largest=True)  # [B, left_tokens] , sorted=True
+        _, idx = torch.topk(cls_attn, max(1, round(N * reduction_ratio)), dim=1, largest=True)  # [B, left_tokens] , sorted=True
 
 
         # # # print("idx: ", idx)
